libs/@blockprotocol/graph/src/shared/stdlib/subgraph/element/entity.ts

```diff
@@ -40,6 +40,56 @@ export const getEntities = <Temporal extends boolean>(
   });
 };
 
+const getRevisionsForEntity = <Temporal extends boolean>(
+  subgraph: Subgraph<Temporal>,
+  entityId: EntityId,
+): Vertices<Temporal>[string] | undefined => {
+  const entityRevisions = subgraph.vertices[entityId];
+
+  if (entityRevisions) {
+    return entityRevisions;
+  }
+
+  // check for the presence of draft versions of the entity in the subgraph
+  if (entityId.split("~")[2]) {
+    /**
+     * This entityId contains a draftId, and so we would have already found it via vertices[entityId] if it exists
+     */
+    return undefined;
+  }
+
+  /**
+   * We haven't found the exact entityId in the subgraph, but it might be qualified by a draftId
+   * – check for vertices which are keyed by a qualified version of the provided entityId
+   */
+  const draftEntityIds = Object.keys(subgraph.vertices).filter((id) =>
+    id.startsWith(`${entityId}~`),
+  );
+
+  if (draftEntityIds.length > 0) {
+    /**
+     * Return a combined version of all the draft editions of this entity present in the subgraph.
+     * Return a combined object of all the draft editions of this entity present in the subgraph.
```

Comment:
Is this duplicative?

```suggestion
     * Return a combined object of all the draft editions of this entity present in the subgraph.
```

Reply:
Yes, copy/paste error – fixed in https://github.com/hashintel/hash/pull/4166/commits/ffdf60640d711be191da268b42ff6b45b7c42c6c